### DISCUSSION TITLE
Needs après discussion avec la cheffe

### DISCUSSION THREAD

**Discussion by Amustache** [CLICGameStar/NiceLAN#1] [General]
C'est très brouillon mais au moins c'est centralisé.

Ne pas hésiter à éditer le OP au fur et à mesure avec les issues.

## Définitions préliminaires
* Une activité, ça correspond à soit un tournoi, soit une animation, soit un autre truc.
* Un partaker, ça correspond soit à un.e joueur.euse, soit à une équipe

## Needs

* Au niveau des users pour l'application, il y a (#34) :
    * Participant.e.s (ne peuvent rien changer, read-only)
    * Gestionnaire d'activité (peuvent modifier le score de quelqu'un sur une activité, ajouter des points, register des gens dans une activité possibilité d'ajouter des points sur toutes les animations)
    * Gestionnaire des activités (peuvent créer des activités, changer des activités, ajouter/enlever des points)
    * Administrateur.trice.s (gérer l'événement, accepter/refuser des inscriptions, etc.)
    * Superadmin (stack + rollbacks + black magic)
* Comptage des points, comptage par player, comptage par team (#35)
* Classement par tournoi, pas besoin de classement par animation (#35)
    * Par animation, on peut avoir un nombre de points différents
* Les teams sont uniques par tournoi (#19)
* Classement par participant.e global (scores de tournois + scores d'animations) (#35)
* Classement par grande team (scores de tournois + scores d'animations) (#35)
* Tous ces trucs peuvent être retrivable par n'importe qui dans le système par telegram, discord, api (#26)
* Possibilité de retrive des informations par activité
* Pour un tournoi :
    * Possibilité d'entrer tous les participants (#36)
    * Donner le format du tournoi (e.g., arbre avec looser bracket) (#37)
    * Création du tournoi (#14)
    * Modification du tournoi (lui a gagné, donc lui passe à la manche suivante)
    * Génération des arbres au fur et à mesure (#37)
    * Avant que résultats finaux disponibles au niveau des tournois, review par gestionnaire d'activité. si oui, ajout des scores. (#38)
* Informations un peu basiques que les gens puissent retrieve (e.g., c'est quand qu'on bouffe, c'est quand que X) (#25)
* iInterfaces :
    * Telegram pour tous les animateur.trice.s des activités, "lui a gagné deux points", feedbacks (vérifier que le pseudo est bien entré, "did you mean")
* Cartes NFC, quand tu scan la carte, ça fait "je veux ajouter des points à ce user" (#2)
    * Ouvre un menu, un bouton pour voir les points, et si t'es admin, un autre bouton pour ajouter des points
* Tournois, les joueurs peuvent autoreport "j'ai gagné" ou "j'ai perdu"
    * Si conflit, anim choisit
    * Régie, où en est le reporting, et qui a report quoi, sur un écran, "machin a report ça, machin a report ça", possibilité de le voir en un coup d'oeil
* Toujours avoir un bouton "je viens de me tromper" et ça rollback (#24)
* Liste de points possibles associés par animation (#22), liste des points par classement pour tournois (#23)
* Possibiltié d'ajouter des points arbitraires aux partakers (#21)
* Log everything au cas où lol (#20)
* Préparer un "protocole panik" si le truc casse en cours de route

**Comment by Wfiles** [comment]
"Possibiltié d'ajouter des points arbitraires aux partakers" dans quelle circonstance on voudrait faire ça

**Reply by Amustache** [reply]
Lors d'happenings par exemple

**Comment by LaClimSx** [comment]
Voilà les 4 systèmes de points pour les tournois :
Solo : 
        -Mode Classique : de 8ème à 5ème -> 6 points ; de 4ème à 3ème -> 10 points ; 2ème -> 14 points ; 1er -> 20 points
        -Mode "Précis" : 8ème -> 6 points ; 7ème -> 8 points ; 6ème -> 10 points ; 5ème -> 12 points ; 4ème -> 14 points  ; 3ème -> 16 points  ; 2ème -> 18 points  ; 1er -> 20 points

Équipes : 
       -Mode classique : 4ème à 3ème -> 4 points ; 2ème -> 8 points ; 1er -> 12 points
       -Mode "précis" : 4ème -> 6 points ; 3ème -> 8 points ; 2ème -> 10 points ; 1er -> 12 points

Toutes les autres places du classement (en solo et équipes) rapportent 1 point de participation
